attackchains/attackchainhandler.go

```diff
@@ -154,23 +154,41 @@ func (h *AttackChainsEngine) DetectAllAttackChainsFromLists(postureResourceSumma
 	var attackChains []*armotypes.AttackChain
 
 	for i := range postureResourceSummaries {
-		attackTracks, err := h.DetectAllAttackChains(postureResourceSummaries[i], vuls[i])
-		if err != nil {
-			return nil, err
-		}
+		for j := range vuls {
+			if h.workLoadMatch(postureResourceSummaries[i], vuls[j]) {
+				attackTracks, err := h.DetectAllAttackChains(postureResourceSummaries[i], vuls[j])
+				if err != nil {
+					return nil, err
+				}
```

Comment:
Let's print an error if workLoad doesn't match.

Reply:
I don't think it's an error if they don't match, we're not expecting the vuln and postureResourceSummary to always be in the same index in their respective lists